packages/shared/watchDebounced/index.test.ts

```diff
@@ -42,6 +42,33 @@ describe('watchDebounced', () => {
     num.value = 5
     await vi.advanceTimersByTimeAsync(75)
     expect(cb).toHaveBeenCalledTimes(2)
-    expect(cb).toHaveBeenCalledWith(4, 2, expect.anything())
+    expect(cb).toHaveBeenCalledWith(5, 4, expect.anything())
```

Comment:
as far as i can tell, this test was already showing the bug but we just rolled with it

second pair of eyes on this test would be helpful